File: SensoroSDKActive-ObjC/SensoroSDKActive-ObjC/ViewController.m
//
//  ViewController.m
//  SensoroSDKActive-ObjC
//
//  Created by skyming on 15/4/10.
//  Copyright (c) 2015年 Sensoro. All rights reserved.
//

#import "ViewController.h"
#import "SENLocationManager.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIButton *actionButton;
@property (weak, nonatomic) IBOutlet UIImageView *image;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    if ([SENLocationManager sharedInstance].started) {
        [_actionButton setTitle:@"结束检测" forState:UIControlStateNormal];
    }else{
        [_actionButton setTitle:@"启动检测" forState:UIControlStateNormal];
    }
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (IBAction)startMonitor:(id)sender {
    
    if (![SENLocationManager sharedInstance].started) {
        [_actionButton setTitle:@"结束检测" forState:UIControlStateNormal];
        [[SENLocationManager sharedInstance]startMonitor:false];
    }else{
        [_actionButton setTitle:@"启动检测" forState:UIControlStateNormal];
        [[SENLocationManager sharedInstance]stopMonitor];
    }
    
}

- (IBAction)saveToAlbum:(id)sender {
    UIImageWriteToSavedPhotosAlbum(_image.image, self, @selector(image:didFinishSavingWithError:contextInfo:), nil);
}

- (void)image:(UIImage *)image didFinishSavingWithError:(NSError *)error contextInfo:(void *)contextInfo{
    
    if (error == nil) {
        
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"提示" message:@"保存成功" delegate:self cancelButtonTitle:nil otherButtonTitles:@"OK", nil];
        [alert show];
        
    }else{
        
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"提示" message:@"保存失败" delegate:self cancelButtonTitle:nil otherButtonTitles:@"OK", nil];
        [alert show];
    }
    
}

@end

File: SensoroSDKActive-Swift/SensoroSDKActive-iOS/ViewController.swift
//
//  ViewController.swift
//  SensoroSDKActive-iOS
//
//  Created by David Yang on 15/4/8.
//  Copyright (c) 2015年 Sensoro. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var image: UIImageView!
    @IBOutlet weak var actionButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        // Do any additional setup after loading the view, typically from a nib.
        if SENLocationManager.sharedInstance.started == true{
            actionButton.setTitle("结束监测", for: .normal);
        }else{
            actionButton.setTitle("启动监测", for: .normal);
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func star(_ sender: AnyObject) {
        
        if SENLocationManager.sharedInstance.started == false{
            actionButton.setTitle("结束监测", for: .normal);
            SENLocationManager.sharedInstance.startMonitor(relaunch: false);
        }else{
            actionButton.setTitle("启动监测", for: .normal);
            SENLocationManager.sharedInstance.stopMonitor();
        }
    }

    @IBAction func saveToAlbum(sender: AnyObject) {
        UIImageWriteToSavedPhotosAlbum(image.image!,
                                       self,#selector(image(image:didFinishSavingWithError:contextInfo:)),nil);
    }
    
    //
    @objc func image(image : UIImage, didFinishSavingWithError error : NSError!, contextInfo info: UnsafeRawPointer) {
        if error == nil {
            let alert = UIAlertView(title: "提示", message: "保存成功", delegate: nil, cancelButtonTitle: "OK");
            alert.show();
        }else{
            let alert = UIAlertView(title: "提示", message: "保存失败", delegate: nil, cancelButtonTitle: "OK");
            alert.show();
        }
    }
}

